builder/swig/csharp.bzl

```diff
@@ -0,0 +1,241 @@
+                ctx.attr._swig.data_runfiles.files,
+        ]),
+        outputs = [wrap_cxx, wrap_csharp] + swig_headers,
+        executable = ctx.file._swig,
+        arguments = args,
+    )
+
+    lib_compilation_context = ctx.attr.lib[CcInfo].compilation_context
+    compilation_context = cc_common.create_compilation_context(
+        headers = depset(swig_headers, transitive = [lib_compilation_context.headers]),
+        defines = lib_compilation_context.defines,
+        framework_includes = lib_compilation_context.framework_includes,
+        includes = depset(
+            [h.dirname for h in ctx.attr.lib[CcInfo].compilation_context.headers.to_list()],
+            transitive = [lib_compilation_context.includes],
+        ),
+        local_defines = lib_compilation_context.local_defines,
+        quote_includes = lib_compilation_context.quote_includes,
+        system_includes = depset(
+            [],
+            transitive = [lib_compilation_context.system_includes],
+        )
+    )
+
+    return [
+        DefaultInfo(files = depset([wrap_cxx, wrap_csharp])),
+        OutputGroupInfo(
+            csharp_source = depset([wrap_csharp]),
+            cxx_source = depset([wrap_cxx]),
+        ),
+        CcInfo(
+            compilation_context = compilation_context,
+            linking_context = ctx.attr.lib[CcInfo].linking_context,
+        ),
+    ]
+
+
+swig_csharp_wrapper = rule(
+    implementation = _swig_csharp_wrapper_impl,
+    attrs = {
+        "lib": attr.label(
+            doc = "The C library for which to generate the wrapper",
+            providers = [CcInfo],
+            mandatory = True,
+        ),
+        "class_name": attr.string(
+            doc = "Optional override for the C# class name (default: same as target name)",
+        ),
+        "interface": attr.label(
+            doc = "Optional SWIG interface (.i) file",
+            allow_single_file = True,
+        ),
+        "includes": attr.label_list(
+            doc = "Additional SWIG files required for wrapper generation",
+            allow_files = True,
+        ),
+        "namespace": attr.string(
+            doc = "C# namespace where to put the sources",
+            mandatory = True,
+        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = False,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+
+def _csharp_native_library(ctx):
+    return [
+        DefaultInfo(
+            files = depset([ctx.file.native_lib]),
+            runfiles = ctx.runfiles(files = [ctx.file.native_lib])
+        ),
+        DotnetAssemblyCompileInfo(
```

Comment:
We need to put DotnetAssembly*Info to build our test (specifically BDD test) targets, which require this information for dependencies.